In [1]:
import pandas as pd
data = pd.read_parquet('/drive2/phatnt/zTrans/data/migration_others_method_no_code.parquet')

In [ ]:
data = data[(data['startCommit'] == 'd59c7d6f4485bdd578ee5c45dfc421a705d48bc0') & (data['endCommit'] == '2a903b90e3a5fb015d59ccef4ccdde265b1746d8')]

In [2]:
data_group_by_migration = data.groupby(['startCommit','endCommit']).agg({'id' : 'count'}).rename(columns = {'id' : 'count'}).sort_values('count', ascending=False)
data_group_by_migration

count
startCommit                              endCommit                                      
5b6305c462e769f6af7c5692ea2f6b187ad80780 f9c9074d89edd036d8c18caebdb4a85a040513d7  17049
7d603c302dccccfa80d8572a2825142431908eb0 191354e0378b4dd2257a6368ba7159348a651409  15452
d59c7d6f4485bdd578ee5c45dfc421a705d48bc0 2a903b90e3a5fb015d59ccef4ccdde265b1746d8  13022
d68640c5a7995fbdbe0aa61dc88b0f0b23b6d5fe 3cfa7b24ee562abfabdf19bc1035c61c3ed3eff2   8422
0e092deee046995b2813e4a130cb61592d9efd84 3cfa7b24ee562abfabdf19bc1035c61c3ed3eff2   7997
...                                                                                  ...
7d0ef8ccbbd32a4985fe20408cc85a88be7d2d7d 7d0ef8ccbbd32a4985fe20408cc85a88be7d2d7d      1
127495197cbd1d1e465c4de34530930964d49b98 127495197cbd1d1e465c4de34530930964d49b98      1
104e7be5542a48bcfb4be78f35e247407f0279be 104e7be5542a48bcfb4be78f35e247407f0279be      1
12b3260071b94f66c078ca4bfef07fe8d28fdea7 12b3260071b94f66c078ca4bfef07fe8d28fdea7      1
1284f0da311a4494221111ff8d4821ca62fdad79 1284f0da311a4494221111ff8d4821ca62fdad79      1

[1041 rows x 1 columns]

In [3]:
data_group_by_migration['count'].describe()

count     1041.000000
mean       165.013449
std        987.206766
min          1.000000
25%          4.000000
50%         15.000000
75%         50.000000
max      17049.000000
Name: count, dtype: float64

In [2]:
filtered_df = data[(data['method_before'].str.len() > 0) & (data['method_after'].str.len() > 0)]
filtered_df

id                                         fromLib  \
0            0                                         ant:ant   
1            1                                         ant:ant   
2            2                                         ant:ant   
3            3                                         ant:ant   
4            4                                         ant:ant   
...        ...                                             ...   
171773  171773  ru.yandex.qatools.properties:properties-loader   
171774  171774  ru.yandex.qatools.properties:properties-loader   
171775  171775  ru.yandex.qatools.properties:properties-loader   
171776  171776  ru.yandex.qatools.properties:properties-loader   
171778  171778                                 xstream:xstream   

                                   toLib  \
0                     org.apache.ant:ant   
1                     org.apache.ant:ant   
2                     org.apache.ant:ant   
3                     org.apache.ant:ant   
4                     org.apache.ant:ant   
...                                  ...   
171773     ru.qatools.commons:properties   
171774     ru.qatools.commons:properties   
171775     ru.qatools.commons:properties   
171776     ru.qatools.commons:properties   
171778  com.thoughtworks.xstream:xstream   

                                              repoName  \
0                                  bobmcwhirter_drools   
1                                  bobmcwhirter_drools   
2                                  bobmcwhirter_drools   
3                                  bobmcwhirter_drools   
4                                  bobmcwhirter_drools   
...                                                ...   
171773  yandex-qatools_hamcrest-pojo-matcher-generator   
171774  yandex-qatools_hamcrest-pojo-matcher-generator   
171775  yandex-qatools_hamcrest-pojo-matcher-generator   
171776  yandex-qatools_hamcrest-pojo-matcher-generator   
171778                                      apache_ace   

                                      prevCommit  \
0       2ba5d35fb486c4a16b5b8b15fc247e8759bfa54f   
1       2ba5d35fb486c4a16b5b8b15fc247e8759bfa54f   
2       2ba5d35fb486c4a16b5b8b15fc247e8759bfa54f   
3       2ba5d35fb486c4a16b5b8b15fc247e8759bfa54f   
4       2ba5d35fb486c4a16b5b8b15fc247e8759bfa54f   
...                                          ...   
171773  13ac1e25cc715b6855890070099a3cdbea36cf56   
171774  13ac1e25cc715b6855890070099a3cdbea36cf56   
171775  13ac1e25cc715b6855890070099a3cdbea36cf56   
171776  13ac1e25cc715b6855890070099a3cdbea36cf56   
171778  5ddf7366ae99b563ef45abb4bdb702459b3dda61   

                                     startCommit  \
0       92384035a8651b675c82689b24837eb8adb81d66   
1       92384035a8651b675c82689b24837eb8adb81d66   
2       92384035a8651b675c82689b24837eb8adb81d66   
3       92384035a8651b675c82689b24837eb8adb81d66   
4       92384035a8651b675c82689b24837eb8adb81d66   
...                                          ...   
171773  cc7a474d3ea53164d28b4a1b23c25989982e7d03   
171774  cc7a474d3ea53164d28b4a1b23c25989982e7d03   
171775  cc7a474d3ea53164d28b4a1b23c25989982e7d03   
171776  cc7a474d3ea53164d28b4a1b23c25989982e7d03   
171778  3df4226489929e6ab565f99481826acd9e7a0b4f   

                                       endCommit  \
0       0df25aead759b3b542a685ac21a3a009bcd22fe9   
1       0df25aead759b3b542a685ac21a3a009bcd22fe9   
2       0df25aead759b3b542a685ac21a3a009bcd22fe9   
3       0df25aead759b3b542a685ac21a3a009bcd22fe9   
4       0df25aead759b3b542a685ac21a3a009bcd22fe9   
...                                          ...   
171773  cc7a474d3ea53164d28b4a1b23c25989982e7d03   
171774  cc7a474d3ea53164d28b4a1b23c25989982e7d03   
171775  cc7a474d3ea53164d28b4a1b23c25989982e7d03   
171776  cc7a474d3ea53164d28b4a1b23c25989982e7d03   
171778  3df4226489929e6ab565f99481826acd9e7a0b4f   

                                                 fileName  \
0       drools-compiler/src/main/java/org/drools/rule/...

In [3]:
special_df = data[~data.index.isin(filtered_df.index)]
special_df

id                                         fromLib  \
14          14                                         ant:ant   
15          15                                         ant:ant   
16          16                                         ant:ant   
22          22                                         ant:ant   
23          23                                         ant:ant   
...        ...                                             ...   
171760  171760  ru.yandex.qatools.properties:properties-loader   
171761  171761  ru.yandex.qatools.properties:properties-loader   
171762  171762  ru.yandex.qatools.properties:properties-loader   
171764  171764  ru.yandex.qatools.properties:properties-loader   
171777  171777  ru.yandex.qatools.properties:properties-loader   

                                toLib  \
14                 org.apache.ant:ant   
15                 org.apache.ant:ant   
16                 org.apache.ant:ant   
22                 org.apache.ant:ant   
23                 org.apache.ant:ant   
...                               ...   
171760  ru.qatools.commons:properties   
171761  ru.qatools.commons:properties   
171762  ru.qatools.commons:properties   
171764  ru.qatools.commons:properties   
171777  ru.qatools.commons:properties   

                                              repoName  \
14                                 bobmcwhirter_drools   
15                                 bobmcwhirter_drools   
16                                 bobmcwhirter_drools   
22                                 bobmcwhirter_drools   
23                                 bobmcwhirter_drools   
...                                                ...   
171760  yandex-qatools_hamcrest-pojo-matcher-generator   
171761  yandex-qatools_hamcrest-pojo-matcher-generator   
171762  yandex-qatools_hamcrest-pojo-matcher-generator   
171764  yandex-qatools_hamcrest-pojo-matcher-generator   
171777  yandex-qatools_hamcrest-pojo-matcher-generator   

                                      prevCommit  \
14      2ba5d35fb486c4a16b5b8b15fc247e8759bfa54f   
15      2ba5d35fb486c4a16b5b8b15fc247e8759bfa54f   
16      2ba5d35fb486c4a16b5b8b15fc247e8759bfa54f   
22      2ba5d35fb486c4a16b5b8b15fc247e8759bfa54f   
23      2ba5d35fb486c4a16b5b8b15fc247e8759bfa54f   
...                                          ...   
171760  13ac1e25cc715b6855890070099a3cdbea36cf56   
171761  13ac1e25cc715b6855890070099a3cdbea36cf56   
171762  13ac1e25cc715b6855890070099a3cdbea36cf56   
171764  13ac1e25cc715b6855890070099a3cdbea36cf56   
171777  13ac1e25cc715b6855890070099a3cdbea36cf56   

                                     startCommit  \
14      92384035a8651b675c82689b24837eb8adb81d66   
15      92384035a8651b675c82689b24837eb8adb81d66   
16      92384035a8651b675c82689b24837eb8adb81d66   
22      92384035a8651b675c82689b24837eb8adb81d66   
23      92384035a8651b675c82689b24837eb8adb81d66   
...                                          ...   
171760  cc7a474d3ea53164d28b4a1b23c25989982e7d03   
171761  cc7a474d3ea53164d28b4a1b23c25989982e7d03   
171762  cc7a474d3ea53164d28b4a1b23c25989982e7d03   
171764  cc7a474d3ea53164d28b4a1b23c25989982e7d03   
171777  cc7a474d3ea53164d28b4a1b23c25989982e7d03   

                                       endCommit  \
14      0df25aead759b3b542a685ac21a3a009bcd22fe9   
15      0df25aead759b3b542a685ac21a3a009bcd22fe9   
16      0df25aead759b3b542a685ac21a3a009bcd22fe9   
22      0df25aead759b3b542a685ac21a3a009bcd22fe9   
23      0df25aead759b3b542a685ac21a3a009bcd22fe9   
...                                          ...   
171760  cc7a474d3ea53164d28b4a1b23c25989982e7d03   
171761  cc7a474d3ea53164d28b4a1b23c25989982e7d03   
171762  cc7a474d3ea53164d28b4a1b23c25989982e7d03   
171764  cc7a474d3ea53164d28b4a1b23c25989982e7d03   
171777  cc7a474d3ea53164d28b4a1b23c25989982e7d03   

                                                 fileName  \
14      drools-compiler/src/test/java/org/drools/reteo...   
15      drools-compiler/src/test

In [4]:
tmp = []
special_cases_before = special_df[(special_df['method_before'].str.len() > 0) & (special_df['method_after'].str.len() == 0)]
special_cases_after = special_df[ (special_df['method_before'].str.len() == 0) & (special_df['method_after'].str.len() > 0)]
# for index,method_be in special_cases_before.iterrows():
#     if method_be['method_before_signature'] in list(special_cases_after['method_after_signature']):
#         if method_be['class_before'] in list(special_cases_after['class_after']):
#             print(method_be['method_before_signature'],'   ',method_be['class_before'],'   ',method_be['startCommit'],'    ',method_be['endCommit'],'   ',method_be['repoName'])
#             print('------>')
# print('*'*15)
# for index,method_af in special_cases_after.iterrows():
#     print(method_af['method_after_signature'],'   ',method_af['class_after'],'   ',method_af['startCommit'],'    ',method_af['endCommit'],'   ',method_af['repoName'])
#     print('--->')


# remain_before_methods_df_by_signature_class = special_cases_before[
#     special_cases_before.apply(
#         lambda row: (
#             row['class_before'],
#             row['method_before_signature'],
#             row['startCommit'],
#             row['endCommit'],
#             row['repoName']
#         ) in valid_pairs,
#         axis=1
#     )
# ]
# print(len(remain_before_methods_df_by_signature_class))
# for method in remain_before_methods_df_by_signature_class.iterrows():
#     print(method['method_before_signature'])

In [8]:
print(special_cases_before.columns)

Index(['id', 'fromLib', 'toLib', 'repoName', 'prevCommit', 'startCommit',
       'endCommit', 'fileName', 'startCommitChanges', 'endCommitChanges',
       'total_methods_before', 'total_methods_after', 'class_before',
       'class_after', 'method_before_name', 'method_after_name',
       'method_before', 'method_after', 'method_diff',
       'method_before_modifiers', 'method_after_modifiers',
       'method_before_return_type', 'method_after_return_type',
       'method_before_parameters', 'method_after_parameters',
       'method_before_signature', 'method_after_signature',
       'method_before_signature_no_mod', 'method_after_signature_no_mod'],
      dtype='object')


In [6]:

valid_pairs = set(zip(
    # special_cases_after['class_after'],
    special_cases_after['method_after_name'],
    special_cases_after['startCommit'],
    special_cases_after['endCommit'],
    special_cases_after['repoName']
))
remain_before_methods_df_by_signature_class = special_cases_before[
    special_cases_before.apply(
        lambda row: (
            # row['class_before'],
            row['method_before_signature'],
            row['startCommit'],
            row['endCommit'],
            row['repoName'],
        ) in valid_pairs,
        axis = 1
    )
]
removed_before_methods_df_by_signature_class: pd.DataFrame = special_cases_before[
    ~(
        special_cases_before.index.isin(remain_before_methods_df_by_signature_class.index)
    )
]
remain_after_methods_df_by_signature_class: pd.DataFrame = special_cases_after[
    special_cases_after.apply(
        lambda row: (
            # row['class_after'],
            row['method_after_signature'],
            row['startCommit'],
            row['endCommit'],
            row['repoName'],
        ) in valid_pairs,
        axis = 1
    )
]
added_after_methods_df_by_signature_class: pd.DataFrame = special_cases_after[
    ~(
        special_cases_after.index.isin(remain_after_methods_df_by_signature_class.index)
    )
]
# added_after_methods_df_by_signature_class: pd.DataFrame = special_cases_after[
#     ~(
#         special_cases_after['method_after_signature'].isin(remain_before_methods_df_by_signature_class['method_before_signature']) &
#         (special_cases_after['class_after'].isin(special_cases_before['class_before'])) &
#         special_cases_after['repoName'].isin(remain_before_methods_df_by_signature_class['repoName']) &
#         special_cases_after['startCommit'].isin(remain_before_methods_df_by_signature_class['startCommit']) &
#         special_cases_after['endCommit'].isin(remain_before_methods_df_by_signature_class['endCommit'])
#     )
# ]
# remain_after_methods_df_by_signature_class: pd.DataFrame = special_cases_after[
#     ~(
#         special_cases_after.index.isin(added_after_methods_df_by_signature_class.index)
#     )
# ]

print('filtered with same repo and migration')
print(f'removed before methods filtered by signature + class: {len(removed_before_methods_df_by_signature_class)}')
print(f'remain before methods filtered by signature + class: {len(remain_before_methods_df_by_signature_class)}')
print()
print(f'added after methods filtered by signature + class: {len(added_after_methods_df_by_signature_class)}')
print(f'remain after methods filtered by signature + class: {len(remain_after_methods_df_by_signature_class)}')
print('-' * 20)

print(f'percentage removed to the before set: {len(removed_before_methods_df_by_signature_class) / len(special_cases_before)}')
print(f'percentage removed to the after set: {len(removed_before_methods_df_by_signature_class) / len(special_cases_after)}')
print()
print(f'percentage added to the before set: {len(added_after_methods_df_by_signature_class) / len(special_cases_before)}')
print(f'percentage added to the after set: {len(added_after_methods_df_by_signature_class) / len(special_cases_after)}')

unique_repos = removed_before_methods_df_by_signature_class['repoName'].unique()
removed_wtf = []
added_wtf = []
removed_filtered = []
added_filtered = []
cutoff = 500


filtered with same repo and migration
removed before methods filtered by signature + class: 51532
remain before methods filtered by signature + class: 0

added after methods filtered by signature + class: 64289
remain after methods filtered by signature + class: 0
--------------------
percentage removed to the before set: 1.0
percentage removed to the after set: 0.8015679198618737

percentage added to the before set: 1.2475549173329195
percentage added to the after set: 1.0


In [10]:
remain_after_methods_df_by_signature_class

id                              fromLib  \
2615      2615                              ant:ant   
7143      7143                              ant:ant   
7649      7649                              ant:ant   
7802      7802                              ant:ant   
7831      7831                              ant:ant   
...        ...                                  ...   
125960  125960               net.sf.trove4j:trove4j   
139587  139587               net.sf.trove4j:trove4j   
151140  151140  org.eclipse.jetty:jetty-annotations   
153579  153579  org.eclipse.jetty:jetty-annotations   
154551  154551  org.eclipse.jetty:jetty-annotations   

                                               toLib               repoName  \
2615                              org.apache.ant:ant          mulesoft_mule   
7143                              org.apache.ant:ant          mulesoft_mule   
7649                              org.apache.ant:ant          mulesoft_mule   
7802                              org.apache.ant:ant          mulesoft_mule   
7831                              org.apache.ant:ant          mulesoft_mule   
...                                              ...                    ...   
125960                         com.carrotsearch:hppc           owlcs_owlapi   
139587                         com.carrotsearch:hppc           owlcs_owlapi   
151140  org.eclipse.jetty.websocket:websocket-server  eclipse_jetty.project   
153579  org.eclipse.jetty.websocket:websocket-server  eclipse_jetty.project   
154551  org.eclipse.jetty.websocket:websocket-server  eclipse_jetty.project   

                                      prevCommit  \
2615    7d72920ff408ab40f8342998536cd46a3f97e82e   
7143    7d72920ff408ab40f8342998536cd46a3f97e82e   
7649    7d72920ff408ab40f8342998536cd46a3f97e82e   
7802    7d72920ff408ab40f8342998536cd46a3f97e82e   
7831    7d72920ff408ab40f8342998536cd46a3f97e82e   
...                                          ...   
125960  cbf63f7f9e3de211999c97955a6bed9add474736   
139587  cbf63f7f9e3de211999c97955a6bed9add474736   
151140  aa22952368532b04565c411d8eecc30d306c73c4   
153579  b0064961985374f1eec5720a52abcc715782914e   
154551  b0064961985374f1eec5720a52abcc715782914e   

                                     startCommit  \
2615    d59c7d6f4485bdd578ee5c45dfc421a705d48bc0   
7143    d59c7d6f4485bdd578ee5c45dfc421a705d48bc0   
7649    d59c7d6f4485bdd578ee5c45dfc421a705d48bc0   
7802    d59c7d6f4485bdd578ee5c45dfc421a705d48bc0   
7831    d59c7d6f4485bdd578ee5c45dfc421a705d48bc0   
...                                          ...   
125960  7d603c302dccccfa80d8572a2825142431908eb0   
139587  7d603c302dccccfa80d8572a2825142431908eb0   
151140  f8950b64bdd7981dbad1f127151100765ffda1f9   
153579  4f6732a7e9ac5df843990d6141955d71a5f633bd   
154551  4f6732a7e9ac5df843990d6141955d71a5f633bd   

                                       endCommit  \
2615    2a903b90e3a5fb015d59ccef4ccdde265b1746d8   
7143    2a903b90e3a5fb015d59ccef4ccdde265b1746d8   
7649    2a903b90e3a5fb015d59ccef4ccdde265b1746d8   
7802    2a903b90e3a5fb015d59ccef4ccdde265b1746d8   
7831    2a903b90e3a5fb015d59ccef4ccdde265b1746d8   
...                                          ...   
125960  8bac5ab5d2ce4d7660c40ef5d337cfb4a8713064   
139587  191354e0378b4dd2257a6368ba7159348a651409   
151140  f8950b64bdd7981dbad1f127151100765ffda1f9   
153579  d85662bd6e2a56bbbb8f753fede2036dfeba740f   
154551  d85662bd6e2a56bbbb8f753fede2036dfeba740f   

                                                 fileName  \
2615     core/src/main/java/org/mule/api/MuleContext.java   
7143    examples/loanbroker-legacy/esb/src/test/java/o...   
7649    modules/cxf/src/test/java/org/mule/module/cxf/...   
7802    modules/cxf/src/test/java/org/mule/module/cxf/...   
7831    modules/cxf/src/test/java/org/mule/module/cxf/...   
...                                                   ...   
125960  parsers/src/main/java/org/semanticweb/owlapi/o...   
139587  parsers/src/main/java/org/seman

In [ ]:
remain1 = remain_after_methods_df_by_signature_class.groupby(['startCommit', 'endCommit']).agg({'id': 'count'}) .rename(columns={'id': 'count'}).sort_values('count', ascending=False)

In [ ]:
remain1['count'].describe()

count    128.000000
mean      22.226562
std       63.552349
min        1.000000
25%        2.000000
50%        3.500000
75%       10.250000
max      402.000000
Name: count, dtype: float64

In [ ]:
valid_pairs = remain_after_methods_df_by_signature_class[['startCommit', 'endCommit']].drop_duplicates()
filtered_data_remain_after_and_mapped_method = data[data[['startCommit', 'endCommit']].apply(tuple, axis=1).isin(valid_pairs.apply(tuple, axis=1))]
filtered_data_remain_after_and_mapped_method  = filtered_data_remain_after_and_mapped_method.groupby(['startCommit', 'endCommit']).agg({'id': 'count'}) .rename(columns={'id': 'count'})
filtered_data_remain_after_and_mapped_method['count'].describe()

count      128.000000
mean      1020.187500
std       2647.107135
min          6.000000
25%         54.750000
50%        167.000000
75%        624.750000
max      17049.000000
Name: count, dtype: float64

- 128 migration / 1041 migration = 12,3% chua cac remain after 
- co 22 method remain after / 1020 tong so method = 2,1% trong 1 migration nơi mà có chứa method remain after

In [ ]:
remain2 = remain_before_methods_df_by_signature_class.groupby(['startCommit', 'endCommit']).agg({'id': 'count'}) .rename(columns={'id': 'count'})
remain2['count'].describe()

count    211.000000
mean      19.308057
std       87.225842
min        1.000000
25%        1.000000
50%        3.000000
75%        6.000000
max      941.000000
Name: count, dtype: float64

In [ ]:
valid_pairs = remain_before_methods_df_by_signature_class[['startCommit', 'endCommit']].drop_duplicates()
filtered_data_remain_before_and_mapped_method = data[data[['startCommit', 'endCommit']].apply(tuple, axis=1).isin(valid_pairs.apply(tuple, axis=1))]
filtered_data_remain_before_and_mapped_method  = filtered_data_remain_before_and_mapped_method.groupby(['startCommit', 'endCommit']).agg({'id': 'count'}) .rename(columns={'id': 'count'})
filtered_data_remain_before_and_mapped_method['count'].describe()

count      211.000000
mean       698.545024
std       2109.893943
min          4.000000
25%         37.500000
50%        102.000000
75%        343.500000
max      17049.000000
Name: count, dtype: float64

- 211 migration / 1041 migration = 20% chua cac remain before
- co 20 method remain after / 698 tong so method = 2,8% trong 1 migration nơi mà có chứa method remain before

# Test

In [12]:
statCommit = 'd59c7d6f4485bdd578ee5c45dfc421a705d48bc0'
endCommit = '2a903b90e3a5fb015d59ccef4ccdde265b1746d8'
sample_remain_before = remain_before_methods_df_by_signature_class[(remain_before_methods_df_by_signature_class['startCommit'] == statCommit) & (remain_before_methods_df_by_signature_class['endCommit'] == endCommit)]
sample_remain_after = remain_after_methods_df_by_signature_class[(remain_after_methods_df_by_signature_class['startCommit'] == statCommit) & (remain_after_methods_df_by_signature_class['endCommit'] == endCommit)]
for index,method_be in sample_remain_before.iterrows():
    print(method_be['method_before_signature'],'   ',method_be['class_before'],'   ',method_be['startCommit'],'    ',method_be['endCommit'],'   ',method_be['repoName'])
    print('---->')
print('remain after')
for index,method_af in sample_remain_after.iterrows():
    print(method_af['method_after_signature'],'   ',method_af['class_after'],'   ',method_af['startCommit'],'    ',method_af['endCommit'],'   ',method_af['repoName'])
    print('---->')

String getClusterId()          d59c7d6f4485bdd578ee5c45dfc421a705d48bc0      2a903b90e3a5fb015d59ccef4ccdde265b1746d8     mulesoft_mule
---->
@Override protected String getConfigResources()     ComponentBindingTestCase     d59c7d6f4485bdd578ee5c45dfc421a705d48bc0      2a903b90e3a5fb015d59ccef4ccdde265b1746d8     mulesoft_mule
---->
remain after
String getClusterId()          d59c7d6f4485bdd578ee5c45dfc421a705d48bc0      2a903b90e3a5fb015d59ccef4ccdde265b1746d8     mulesoft_mule
---->
@Override protected String getConfigResources()     TransformersMule2669TestCase     d59c7d6f4485bdd578ee5c45dfc421a705d48bc0      2a903b90e3a5fb015d59ccef4ccdde265b1746d8     mulesoft_mule
---->
@Override protected String getConfigResources()     DatabindingTestCase     d59c7d6f4485bdd578ee5c45dfc421a705d48bc0      2a903b90e3a5fb015d59ccef4ccdde265b1746d8     mulesoft_mule
---->
@Override protected String getConfigResources()     GeneratedClientTestCase     d59c7d6f4485bdd578ee5c45dfc421a705d48bc0      2a